test_conformance/select/test_select.cpp

```diff
@@ -396,111 +388,66 @@ static int doTest(cl_command_queue queue, cl_context context, Type stype, Type c
         cmp_stride = block_elements * step * (0xffffffffffffffffULL / 0x100000000ULL + 1);
 
     log_info("Testing...");
-    d = MTdataHolder(gRandomSeed);
     uint64_t i;
     for (i=0; i < blocks; i+=step)
     {
-        void *s1 = clEnqueueMapBuffer( queue, src1, CL_TRUE, CL_MAP_WRITE, 0, BUFFER_SIZE, 0, NULL, NULL, &err );
-        if( err ){ log_error( "Error: Could not map src1" ); goto exit; }
-        // Setup the input data to change for each block
-        initSrcBuffer( s1, stype, d);
-
-        void *s2 = clEnqueueMapBuffer( queue, src2, CL_TRUE, CL_MAP_WRITE, 0, BUFFER_SIZE, 0, NULL, NULL, &err );
-        if( err ){ log_error( "Error: Could not map src2" ); goto exit; }
-        // Setup the input data to change for each block
-        initSrcBuffer( s2, stype, d);
-
-        void *s3 = clEnqueueMapBuffer( queue, cmp, CL_TRUE, CL_MAP_WRITE, 0, BUFFER_SIZE, 0, NULL, NULL, &err );
-        if( err ){ log_error( "Error: Could not map cmp" ); goto exit; }
-        // Setup the input data to change for each block
-        initCmpBuffer(s3, cmptype, i * cmp_stride, block_elements);
-
-        if( (err = clEnqueueUnmapMemObject( queue, src1, s1, 0, NULL, NULL )))
-        { log_error( "Error: coult not unmap src1\n" );  ++s_test_fail; goto exit; }
-        if( (err = clEnqueueUnmapMemObject( queue, src2, s2, 0, NULL, NULL )))
-        { log_error( "Error: coult not unmap src2\n" );  ++s_test_fail; goto exit; }
-        if( (err = clEnqueueUnmapMemObject( queue, cmp, s3, 0, NULL, NULL )))
-        { log_error( "Error: coult not unmap cmp\n" );  ++s_test_fail; goto exit; }
-
-        // Create the reference result
-        err = clEnqueueReadBuffer(queue, src1, CL_TRUE, 0, BUFFER_SIZE,
-                                  src1_host, 0, NULL, NULL);
-        if (err)
-        {
-            log_error("Error: Reading buffer from src1 to src1_host failed\n");
-            ++s_test_fail;
-            goto exit;
-        }
-        err = clEnqueueReadBuffer(queue, src2, CL_TRUE, 0, BUFFER_SIZE,
-                                  src2_host, 0, NULL, NULL);
-        if (err)
-        {
-            log_error("Error: Reading buffer from src2 to src2_host failed\n");
-            ++s_test_fail;
-            goto exit;
-        }
-        err = clEnqueueReadBuffer(queue, cmp, CL_TRUE, 0, BUFFER_SIZE, cmp_host,
-                                  0, NULL, NULL);
-        if (err)
-        {
-            log_error("Error: Reading buffer from cmp to cmp_host failed\n");
-            ++s_test_fail;
-            goto exit;
-        }
+        initSrcBuffer(src1_host.data(), stype, d);
+        initSrcBuffer(src2_host.data(), stype, d);
+        initCmpBuffer(cmp_host.data(), cmptype, i * cmp_stride, block_elements);
+
+        clEventWrapper user_event = clCreateUserEvent(context, &err);
```

Comment:
As far as I can tell this user event is never used.  Can it be removed?

Reply:
event removed

> I did not see any change in coverage ...

apparently you are right, my measurement must have been skewed, sorry for confusion